src/test/java/io/github/thebusybiscuit/slimefun4/api/events/TestSlimefunBlockPlaceEvent.java

```diff
@@ -0,0 +1,118 @@
+    void testEventIsFired() {
+        Player player = new PlayerMock(server, "SomePlayer");
+
+        World world = server.addSimpleWorld("my_world");
+        ItemStack vanillaItem = new ItemStack(Material.GREEN_TERRACOTTA);
+        Block block = new BlockMock(Material.GREEN_TERRACOTTA, new Location(world, 1, 1, 1));
+
+        Slimefun.getRegistry().getWorlds().put("my_world", new BlockStorage(world));
+        BlockStorage.addBlockInfo(block, "id", "FOOD_COMPOSTER");
+
+        server.getPluginManager().callEvent(new SlimefunBlockPlaceEvent(player, vanillaItem, block, slimefunItem));
+        server.getPluginManager().assertEventFired(SlimefunBlockPlaceEvent.class, e -> true);
+    }
+
+    @Test
+    @DisplayName("Test getting the getters from the event")
+    void testGetters() {
+        Player player = new PlayerMock(server, "SomePlayer");
+        ItemStack itemStack = new ItemStack(Material.GREEN_TERRACOTTA);
+        player.getInventory().setItemInMainHand(itemStack);
+
+        World world = server.addSimpleWorld("my_world");
+        Block block = new BlockMock(Material.GREEN_TERRACOTTA, new Location(world, 1, 1, 1));
+
+        Slimefun.getRegistry().getWorlds().put("my_world", new BlockStorage(world));
+        BlockStorage.addBlockInfo(block, "id", "FOOD_COMPOSTER");
+
+        server.getPluginManager().callEvent(new SlimefunBlockPlaceEvent(player, itemStack, block, slimefunItem));
+        server.getPluginManager().assertEventFired(SlimefunBlockPlaceEvent.class, e -> {
+            Assertions.assertEquals(block, e.getBlockPlaced());
+            Assertions.assertEquals(slimefunItem, e.getSlimefunItem());
+            Assertions.assertEquals(itemStack, e.getItemStack());
+            Assertions.assertEquals(player, e.getPlayer());
+            Assertions.assertFalse(e.isCancelled());
+            return true;
+        });
+    }
+
+    @Test
+    @DisplayName("Test the event cancelling")
+    void testIsCancelled() {
+        server.getPluginManager().registerEvents(new Listener() {
+            @EventHandler
+            public void onBlockPlace(SlimefunBlockPlaceEvent event) {
+                event.setCancelled(true);
+            }
+        }, plugin);
+
+        Player player = new PlayerMock(server, "SomePlayer");
+        ItemStack itemStack = new ItemStack(Material.GREEN_TERRACOTTA);
+        player.getInventory().setItemInMainHand(itemStack);
+
+        World world = server.addSimpleWorld("my_world");
+        Block block = new BlockMock(Material.GREEN_TERRACOTTA, new Location(world, 1, 1, 1));
+
+        Slimefun.getRegistry().getWorlds().put("my_world", new BlockStorage(world));
+        BlockStorage.addBlockInfo(block, "id", "FOOD_COMPOSTER");
+
+        server.getPluginManager().callEvent(new SlimefunBlockPlaceEvent(player, itemStack, block, slimefunItem));
+        server.getPluginManager().assertEventFired(SlimefunBlockPlaceEvent.class, e -> {
+            Assertions.assertTrue(e.isCancelled());
```

Comment:
After the above is fixed, we need to assert that `BlockPlaceEvent ` is cancelled too (same with break).